diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7dfd353..86220d3 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -5,6 +5,9 @@ endif()
 cmake_minimum_required(VERSION 3.0.0)
 project(SDL2 C CXX)

+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
+
 if(WINDOWS_STORE)
   enable_language(CXX)
   cmake_minimum_required(VERSION 3.11)
@@ -1668,7 +1671,12 @@ elseif(APPLE)
     message_error("SDL_FILE must be enabled to build on MacOS X")
   endif()

-  file(GLOB MISC_SOURCES ${SDL2_SOURCE_DIR}/src/misc/macosx/*.m)
+  if(IOS OR TVOS)
+      file(GLOB MISC_SOURCES ${SDL2_SOURCE_DIR}/src/misc/ios/*.m)
+  else()
+      file(GLOB MISC_SOURCES ${SDL2_SOURCE_DIR}/src/misc/macosx/*.m)
+  endif()
+
   set(SOURCE_FILES ${SOURCE_FILES} ${MISC_SOURCES})
   set(HAVE_SDL_MISC TRUE)

@@ -2253,7 +2261,7 @@ endif()

 if(ANDROID)
   if(HAVE_HIDAPI)
-    add_library(hidapi SHARED ${SDL2_SOURCE_DIR}/src/hidapi/android/hid.cpp)
+    add_library(hidapi ${SDL2_SOURCE_DIR}/src/hidapi/android/hid.cpp)
   endif()

   if(MSVC AND NOT LIBC)
